--- gst-plugins-bad-0.10.11.orig/gst-plugins-bad-0.10.11/ext/directfb/dfb-example.c	2009-04-01 13:25:18.000000000 +0200
+++ gst-plugins-bad-0.10.11/gst-plugins-bad-0.10.11/ext/directfb/dfb-example.c	2009-04-01 13:27:44.000000000 +0200
@@ -1,5 +1,5 @@
 
-#include <directfb.h>
+#include <directfb/directfb.h>
 #include <gst/gst.h>
 
 static IDirectFB *dfb = NULL;
--- gst-plugins-bad-0.10.11.orig/gst-plugins-bad-0.10.11/ext/directfb/dfbvideosink.h	2009-04-01 13:25:18.000000000 +0200
+++ gst-plugins-bad-0.10.11/gst-plugins-bad-0.10.11/ext/directfb/dfbvideosink.h	2009-04-01 13:27:44.000000000 +0200
@@ -22,7 +22,7 @@
 
 #include <gst/video/gstvideosink.h>
 
-#include <directfb.h>
+#include <directfb/directfb.h>
 
 G_BEGIN_DECLS
 
--- gst-plugins-bad-0.10.11.orig/gst-plugins-bad-0.10.11/tests/examples/directfb/gstdfb.c	2009-04-01 13:25:18.000000000 +0200
+++ gst-plugins-bad-0.10.11/gst-plugins-bad-0.10.11/tests/examples/directfb/gstdfb.c	2009-04-01 13:26:09.000000000 +0200
@@ -35,7 +35,7 @@
 #include <math.h>
 #include <time.h>
 
-#include <directfb.h>
+#include <directfb/directfb.h>
 #include <gst/gst.h>
 #include <string.h>
 
