Index: mc-4.6.2/src/Makefile.am
===================================================================
--- mc-4.6.2.orig/src/Makefile.am	2009-02-01 19:30:21.000000000 +0000
+++ mc-4.6.2/src/Makefile.am	2009-02-25 13:24:16.000000000 +0000
@@ -14,7 +14,9 @@
 endif
 
 noinst_PROGRAMS = man2hlp
-man2hlp_LDADD = $(GLIB_LIBS)
+man2hlp$(EXEEXT): $(man2hlp_OBJECTS) $(man2hlp_DEPENDENCIES)
+	@rm -f man2hlp$(EXEEXT)
+	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(CFLAGS_FOR_BUILD) $(GLIB_CFLAGS_FOR_BUILD) -I$(top_srcdir) $(GLIB_LIBS_FOR_BUILD) $(man2hlp_SOURCES) -o $@
 
 mcmfmt_SOURCES = mfmt.c
 
