diff -Naur mumudvb.orig/configure.ac mumudvb/configure.ac
--- mumudvb.orig/configure.ac	2009-10-01 22:40:41.000000000 +0200
+++ mumudvb/configure.ac	2009-10-07 22:46:13.000000000 +0200
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ([2.63])
+AC_PREREQ([2.61])
 AC_INIT([MuMuDVB], [1.6], [mumudvb@braice.net])
 AC_CONFIG_SRCDIR([src/mumudvb.c])
 AM_INIT_AUTOMAKE
