diff -auNr Python-2.6.1.orig/Modules/_ctypes/libffi/Makefile.in Python-2.6.1/Modules/_ctypes/libffi/Makefile.in
--- Python-2.6.1.orig/Modules/_ctypes/libffi/Makefile.in	2008-03-04 21:09:11.000000000 +0100
+++ Python-2.6.1/Modules/_ctypes/libffi/Makefile.in	2009-01-16 23:18:34.000000000 +0100
@@ -34,7 +34,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-@MIPS_TRUE@am__append_1 = src/mips/ffi.c src/mips/o32.S src/mips/n32.S
+@MIPS_IRIX_TRUE@am__append_1 = src/mips/ffi.c src/mips/o32.S src/mips/n32.S
 @X86_TRUE@am__append_2 = src/x86/ffi.c src/x86/sysv.S
 @X86_FREEBSD_TRUE@am__append_3 = src/x86/ffi.c src/x86/freebsd.S
 @X86_WIN32_TRUE@am__append_4 = src/x86/ffi.c src/x86/win32.S
@@ -57,6 +57,7 @@
 @SH64_TRUE@am__append_21 = src/sh64/sysv.S src/sh64/ffi.c
 @PA_LINUX_TRUE@am__append_22 = src/pa/linux.S src/pa/ffi.c
 @PA_HPUX_TRUE@am__append_23 = src/pa/hpux32.S src/pa/ffi.c
+@MIPS_LINUX_TRUE@am__append_24 = src/mips/ffi.c src/mips/o32.S
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/doc/stamp-vti \
@@ -416,7 +417,8 @@
 	$(am__append_12) $(am__append_13) $(am__append_14) \
 	$(am__append_15) $(am__append_16) $(am__append_17) \
 	$(am__append_18) $(am__append_19) $(am__append_20) \
-	$(am__append_21) $(am__append_22) $(am__append_23)
+	$(am__append_21) $(am__append_22) $(am__append_23) \
+	$(am__append_24)
 libffi_convenience_la_SOURCES = $(libffi_la_SOURCES)
 nodist_libffi_convenience_la_SOURCES = $(nodist_libffi_la_SOURCES)
 AM_CFLAGS = -Wall -g -fexceptions
diff -auNr Python-2.6.1.orig/Modules/_ctypes/libffi/configure.ac Python-2.6.1/Modules/_ctypes/libffi/configure.ac
--- Python-2.6.1.orig/Modules/_ctypes/libffi/configure.ac	2008-05-23 17:06:50.000000000 +0200
+++ Python-2.6.1/Modules/_ctypes/libffi/configure.ac	2009-01-16 23:10:07.000000000 +0100
@@ -106,10 +106,10 @@
 	;;
 
   mips-sgi-irix5.* | mips-sgi-irix6.*)
-	TARGET=MIPS; TARGETDIR=mips
+	TARGET=MIPS_IRIX; TARGETDIR=mips
 	;;
   mips*-*-linux*)
-	TARGET=MIPS; TARGETDIR=mips
+	TARGET=MIPS_LINUX; TARGETDIR=mips
 	;;
 
   powerpc*-*-linux* | powerpc-*-sysv*)
@@ -162,7 +162,8 @@
   AC_MSG_ERROR(["libffi has not been ported to $host."])
 fi
 
-AM_CONDITIONAL(MIPS, test x$TARGET = xMIPS)
+AM_CONDITIONAL(MIPS_IRIX, test x$TARGET = xMIPS_IRIX)
+AM_CONDITIONAL(MIPS_LINUX, test x$TARGET = xMIPS_LINUX)
 AM_CONDITIONAL(SPARC, test x$TARGET = xSPARC)
 AM_CONDITIONAL(X86, test x$TARGET = xX86)
 AM_CONDITIONAL(X86_FREEBSD, test x$TARGET = xX86_FREEBSD)
