diff -ruN libnl-1.1/lib/route/link/vlan.c libnl-1.1-new/lib/route/link/vlan.c
--- libnl-1.1/lib/route/link/vlan.c	2008-01-14 18:48:45.000000000 +0300
+++ libnl-1.1-new/lib/route/link/vlan.c	2009-01-30 10:55:09.000000000 +0300
@@ -26,7 +26,9 @@
 #include <netlink/route/link/info-api.h>
 #include <netlink/route/link/vlan.h>
 
+#ifndef VLAN_FLAG_REORDER_HDR
 #include <linux/if_vlan.h>
+#endif
 
 /** @cond SKIP */
 #define VLAN_HAS_ID		(1<<0)
--- libnl-1.1.orig/include/netlink-local.h	2008-01-14 16:48:45.000000000 +0100
+++ libnl-1.1/include/netlink-local.h	2009-07-21 16:05:29.000000000 +0200
@@ -26,6 +26,7 @@
 #include <sys/socket.h>
 #include <inttypes.h>
 #include <assert.h>
+#include <limits.h>
 
 #include <arpa/inet.h>
 #include <netdb.h>
