diff -Naur gst-plugins-base-0.10.23.orig/gst/playback/gstplaysink.c gst-plugins-base-0.10.23/gst/playback/gstplaysink.c
--- gst-plugins-base-0.10.23.orig/gst/playback/gstplaysink.c	2009-07-13 23:11:10.000000000 +0200
+++ gst-plugins-base-0.10.23/gst/playback/gstplaysink.c	2009-07-13 23:42:12.000000000 +0200
@@ -1324,6 +1324,7 @@
     GST_DEBUG_OBJECT (playsink, "trying configured audiosink");
     chain->sink = try_element (playsink, playsink->audio_sink, FALSE);
   }
+  elem = NULL;
   if (chain->sink == NULL) {
     GST_DEBUG_OBJECT (playsink, "trying autoaudiosink");
     elem = gst_element_factory_make ("autoaudiosink", "audiosink");
