diff -Naur busybox-1.11.3/miscutils/hdparm.c busybox-1.11.3-hdparm_M/miscutils/hdparm.c
--- busybox-1.11.3/miscutils/hdparm.c	2008-06-25 14:51:29.000000000 +0200
+++ busybox-1.11.3-hdparm_M/miscutils/hdparm.c	2008-12-05 11:04:15.000000000 +0100
@@ -283,6 +283,7 @@
 	smallint set_sleepnow, get_sleepnow;
 	smallint get_powermode;
 	smallint set_apmmode, get_apmmode;
+	smallint set_acoustic, get_acoustic;
 	int xfermode_requested;
 	unsigned long dkeep;
 	unsigned long standby_requested; /* 0..255 */
@@ -292,6 +293,7 @@
 	unsigned long wcache;
 	unsigned long doorlock;
 	unsigned long apmmode;
+	unsigned long acoustic;
 #endif
 	USE_FEATURE_HDPARM_GET_IDENTITY(        smallint get_IDentity;)
 	USE_FEATURE_HDPARM_HDIO_TRISTATE_HWIF(  smallint set_busstate, get_busstate;)
@@ -376,6 +378,8 @@
 #define get_powermode      (G.get_powermode          )
 #define set_apmmode        (G.set_apmmode            )
 #define get_apmmode        (G.get_apmmode            )
+#define set_acoustic       (G.set_acoustic           )
+#define get_acoustic       (G.get_acoustic           )
 #define xfermode_requested (G.xfermode_requested     )
 #define dkeep              (G.dkeep                  )
 #define standby_requested  (G.standby_requested      )
@@ -385,6 +389,7 @@
 #define wcache             (G.wcache                 )
 #define doorlock           (G.doorlock               )
 #define apmmode            (G.apmmode                )
+#define acoustic           (G.acoustic               )
 #define get_IDentity       (G.get_IDentity           )
 #define set_busstate       (G.set_busstate           )
 #define get_busstate       (G.get_busstate           )
@@ -1660,6 +1665,17 @@
 		ioctl_or_warn(fd, HDIO_DRIVE_CMD, &args);
 		args[1] = 0;
 	}
+	if (set_acoustic)
+	{
+		if (get_acoustic)
+			printf(" setting AAM level to 0x%02lX (%ld)\n", acoustic, acoustic);
+		ioctl_or_warn(fd, HDIO_SET_ACOUSTIC, (int*)acoustic);
+	}
+	if (get_acoustic)
+	{
+		ioctl_or_warn(fd, HDIO_GET_ACOUSTIC, (unsigned long*)&parm);
+		printf(" acoustic     = %2ld (128=quiet ... 254=fast)\n", parm);
+	}
 	if (set_wcache)	{
 #ifdef DO_FLUSHCACHE
 #ifndef WIN_FLUSHCACHE
@@ -1963,7 +1979,7 @@
 	USE_FEATURE_HDPARM_GET_IDENTITY("iI")
 	USE_FEATURE_HDPARM_HDIO_GETSET_DMA("d::")
 #ifdef HDIO_DRIVE_CMD
-	"S:D:P:X:K:A:L:W:CyYzZ"
+	"S:D:P:X:K:A:L:W:CyYzZM:"
 #endif
 	USE_FEATURE_HDPARM_HDIO_UNREGISTER_HWIF("U:")
 #ifdef HDIO_GET_QDMA
@@ -2000,6 +2016,7 @@
 		if (c == 'c') parse_opts(&get_io32bit, &set_io32bit, &io32bit, 0, INT_MAX /*8*/);
 		if (c == 'k') parse_opts(&get_keep, &set_keep, &keep, 0, 1);
 		if (c == 'a') parse_opts(&get_readahead, &set_readahead, &Xreadahead, 0, INT_MAX);
+		if (c == 'M') parse_opts(&get_acoustic, &set_acoustic, &acoustic, 1, 254);
 		if (c == 'B') parse_opts(&get_apmmode, &set_apmmode, &apmmode, 1, 255);
 		do_flush |= do_timings |= (c == 't');
 		do_flush |= do_ctimings |= (c == 'T');
