diff --git a/data/setup.xml b/data/setup.xml
index e2ab09b..823fa6f 100644
--- a/data/setup.xml
+++ b/data/setup.xml
@@ -23,6 +23,7 @@
 			<item level="0" text="Margin after record">config.recording.margin_after</item>
 			<item level="0" text="Show blinking clock in display during recording">config.usage.blinking_display_clock_during_recording</item>
 			<item level="2" text="Load Length of Movies in Movielist">config.usage.load_length_of_movies_in_moviellist</item>
+			<item level="1" text="Enable teletext caching">config.usage.enable_tt_caching</item>
 			<item level="1" text="Show positioner movement">config.usage.showdish</item>
 			<item level="1" text="Enable multiple bouquets">config.usage.multibouquet</item>
 			<item level="1" text="Change bouquets in quickzap">config.usage.quickzap_bouquet_change</item>
diff --git a/lib/components/Makefile.am b/lib/components/Makefile.am
index e377ab8..e45ae56 100644
--- a/lib/components/Makefile.am
+++ b/lib/components/Makefile.am
@@ -3,4 +3,4 @@ INCLUDES = \
 
 noinst_LIBRARIES = libenigma_components.a
 
-libenigma_components_a_SOURCES = scan.cpp file_eraser.cpp
+libenigma_components_a_SOURCES = scan.cpp file_eraser.cpp tuxtxtapp.cpp
diff --git a/lib/components/tuxtxtapp.cpp b/lib/components/tuxtxtapp.cpp
new file mode 100644
index 0000000..b08694b
--- /dev/null
+++ b/lib/components/tuxtxtapp.cpp
@@ -0,0 +1,113 @@
+#include <lib/components/tuxtxtapp.h>
+#include <lib/base/init.h>
+#include <lib/base/init_num.h>
+#include <lib/driver/rc.h>
+#include <lib/gdi/lcd.h>
+#include <lib/gdi/fb.h>
+
+extern "C" int tuxtxt_run_ui(int pid, int demux);
+extern "C" int tuxtxt_init();
+extern "C" void tuxtxt_start(int tpid, int demux);
+extern "C" int tuxtxt_stop();
+extern "C" void tuxtxt_close();
+
+eAutoInitP0<eTuxtxtApp> init_eTuxtxtApp(eAutoInitNumbers::lowlevel, "Tuxtxt");
+eTuxtxtApp *eTuxtxtApp::instance;
+
+eTuxtxtApp::eTuxtxtApp() : pid(0), enableTtCaching(false), uiRunning(false)
+{
+	pthread_mutex_init( &cacheChangeLock, 0 );
+	if (!instance)
+		instance=this;
+}
+
+eTuxtxtApp::~eTuxtxtApp()
+{
+	if (instance==this)
+		instance=0;
+	kill();
+	pthread_mutex_destroy( &cacheChangeLock );
+}
+
+int eTuxtxtApp::startUi()
+{
+	if (pid && fbClass::getInstance()->lock() >= 0)
+	{
+		eDBoxLCD::getInstance()->lock();
+		eRCInput::getInstance()->lock();
+		pthread_mutex_lock( &cacheChangeLock );
+		uiRunning = true;
+		pthread_mutex_unlock( &cacheChangeLock );
+		run();
+	}
+	return 0;
+}
+
+void eTuxtxtApp::thread()
+{
+	hasStarted();
+	tuxtxt_run_ui(pid, demux);
+}
+
+void eTuxtxtApp::thread_finished()
+{
+	uiRunning = false;
+	eRCInput::getInstance()->unlock();
+	eDBoxLCD::getInstance()->unlock();
+	eDBoxLCD::getInstance()->update();
+	fbClass::getInstance()->unlock();
+}
+
+void eTuxtxtApp::initCache()
+{
+	if (enableTtCaching)
+		tuxtxt_init();
+}
+
+void eTuxtxtApp::freeCache()
+{
+	pthread_mutex_lock( &cacheChangeLock );
+	if ( !uiRunning )
+	{
+		tuxtxt_close();
+		pid = 0;
+	}
+	pthread_mutex_unlock( &cacheChangeLock );
+}
+
+void eTuxtxtApp::startCaching( int tpid, int tdemux)
+{
+	pid = tpid;
+	demux = tdemux;
+	if (enableTtCaching)
+		tuxtxt_start(pid, demux);
+}
+
+void eTuxtxtApp::stopCaching()
+{
+	pthread_mutex_lock( &cacheChangeLock );
+	if ( !uiRunning )
+		tuxtxt_stop();
+
+	pthread_mutex_unlock( &cacheChangeLock );
+}
+
+void eTuxtxtApp::setEnableTtCachingOnOff( int onoff )
+{
+	if (onoff && !enableTtCaching)		// Switch caching on
+	{
+		enableTtCaching = true;
+		if (pid)
+		{
+			initCache();
+			startCaching(pid, demux);
+		}
+	}
+	else if (!onoff && enableTtCaching)	// Switch caching off
+	{
+		enableTtCaching = false;
+		int savePid = pid;
+		freeCache();
+		pid = savePid;
+	}
+}
diff --git a/lib/components/tuxtxtapp.h b/lib/components/tuxtxtapp.h
new file mode 100644
index 0000000..9cd8f0c
--- /dev/null
+++ b/lib/components/tuxtxtapp.h
@@ -0,0 +1,33 @@
+#ifndef __LIB_COMPONENTS_TUXTXTAPP_H__
+#define __LIB_COMPONENTS_TUXTXTAPP_H__
+
+#include <string>
+#include <lib/base/ebase.h>
+#include <lib/base/thread.h>
+
+class eTuxtxtApp: private eThread, public Object
+{
+#ifndef SWIG
+	int pid;
+	int demux;
+	bool enableTtCaching, uiRunning;
+	static eTuxtxtApp *instance;
+	pthread_mutex_t cacheChangeLock;
+
+	void thread();
+	void thread_finished();
+#endif
+public:
+	eTuxtxtApp();
+	~eTuxtxtApp();
+	static eTuxtxtApp *getInstance() { return instance; }
+	int startUi();
+	void initCache();
+	void freeCache();
+	void startCaching( int tpid, int tdemux );
+	void stopCaching();
+	void resetPid() { pid = 0; demux = 0; }
+	void setEnableTtCachingOnOff( int onoff );
+};
+
+#endif // __LIB_COMPONENTS_TUXTXTAPP_H__
diff --git a/lib/dvb/decoder.cpp b/lib/dvb/decoder.cpp
index 065a49c..173f009 100644
--- a/lib/dvb/decoder.cpp
+++ b/lib/dvb/decoder.cpp
@@ -1,6 +1,7 @@
 #include <lib/base/ebase.h>
 #include <lib/base/eerror.h>
 #include <lib/dvb/decoder.h>
+#include <lib/components/tuxtxtapp.h>
 #if HAVE_DVB_API_VERSION < 3 
 #define audioStatus audio_status
 #define videoStatus video_status
@@ -974,7 +975,11 @@ int eTSMPEGDecoder::setState()
 	if (m_changed & changeText)
 	{
 		if (m_text)
+		{
 			m_text->stop();
+			if ( m_decoder == 0 )   // Tuxtxt caching actions only on primary decoder
+				eTuxtxtApp::getInstance()->stopCaching();
+		}
 		m_text = 0;
 	}
 	if (m_changed & changePCR)
@@ -1015,7 +1020,16 @@ int eTSMPEGDecoder::setState()
 			m_text = new eDVBTText(m_demux);
 			if (m_text->startPid(m_textpid))
 				res = -1;
+
+			if ( m_decoder == 0 )	// Tuxtxt caching actions only on primary decoder
+			{
+				uint8_t demux = 0;
+				m_demux->getCADemuxID(demux);
+				eTuxtxtApp::getInstance()->startCaching(m_textpid, demux);
+			}
 		}
+		else if ( m_decoder == 0 )      // Tuxtxt caching actions only on primary decoder
+			eTuxtxtApp::getInstance()->resetPid();
 		m_changed &= ~changeText;
 	}
 #endif
@@ -1066,10 +1080,16 @@ eTSMPEGDecoder::eTSMPEGDecoder(eDVBDemux *demux, int decoder)
 	demux->connectEvent(slot(*this, &eTSMPEGDecoder::demux_event), m_demux_event_conn);
 	CONNECT(m_showSinglePicTimer->timeout, eTSMPEGDecoder::finishShowSinglePic);
 	m_is_ff = m_is_sm = m_is_trickmode = 0;
+	
+	if ( m_decoder == 0 )	// Tuxtxt caching actions only on primary decoder
+		eTuxtxtApp::getInstance()->initCache();
 }
 
 eTSMPEGDecoder::~eTSMPEGDecoder()
 {
+	if ( m_decoder == 0 )	// Tuxtxt caching actions only on primary decoder
+		eTuxtxtApp::getInstance()->freeCache();
+
 	finishShowSinglePic();
 	m_vpid = m_apid = m_pcrpid = m_textpid = pidNone;
 	m_changed = -1;
diff --git a/lib/python/Components/UsageConfig.py b/lib/python/Components/UsageConfig.py
index 21e057f..f2c864e 100644
--- a/lib/python/Components/UsageConfig.py
+++ b/lib/python/Components/UsageConfig.py
@@ -1,6 +1,6 @@
 from Components.Harddisk import harddiskmanager
 from config import ConfigSubsection, ConfigYesNo, config, ConfigSelection, ConfigText, ConfigNumber, ConfigSet, ConfigLocations
-from enigma import Misc_Options, setTunerTypePriorityOrder;
+from enigma import Misc_Options, setTunerTypePriorityOrder, setEnableTtCachingOnOff
 from SystemInfo import SystemInfo
 import os
 
@@ -17,6 +17,7 @@ def InitUsageConfig():
 	config.usage.show_infobar_on_zap = ConfigYesNo(default = True)
 	config.usage.show_infobar_on_skip = ConfigYesNo(default = True)
 	config.usage.show_infobar_on_event_change = ConfigYesNo(default = True)
+	config.usage.enable_tt_caching = ConfigYesNo(default = True)
 	config.usage.hdd_standby = ConfigSelection(default = "600", choices = [
 		("0", _("no standby")), ("10", "10 " + _("seconds")), ("30", "30 " + _("seconds")),
 		("60", "1 " + _("minute")), ("120", "2 " + _("minutes")),
@@ -59,7 +60,11 @@ def InitUsageConfig():
 
 	config.usage.blinking_display_clock_during_recording = ConfigYesNo(default = False)
 	config.usage.load_length_of_movies_in_moviellist = ConfigYesNo(default = True)
-	
+
+	def EnableTtCachingChanged(configElement):
+		setEnableTtCachingOnOff(int(configElement.value))
+	config.usage.enable_tt_caching.addNotifier(EnableTtCachingChanged)
+
 	def TunerTypePriorityOrderChanged(configElement):
 		setTunerTypePriorityOrder(int(configElement.value))
 	config.usage.alternatives_priority.addNotifier(TunerTypePriorityOrderChanged)
diff --git a/lib/python/enigma_python.i b/lib/python/enigma_python.i
index bdf1b14..358dcfb 100644
--- a/lib/python/enigma_python.i
+++ b/lib/python/enigma_python.i
@@ -86,6 +86,7 @@ is usually caused by not marking PSignals as immutable.
 #include <lib/dvb/pmt.h>
 #include <lib/components/scan.h>
 #include <lib/components/file_eraser.h>
+#include <lib/components/tuxtxtapp.h>
 #include <lib/driver/avswitch.h>
 #include <lib/driver/rfmod.h>
 #include <lib/driver/misc_options.h>
@@ -155,6 +156,9 @@ typedef long time_t;
 %immutable eAVSwitch::vcr_sb_notifier;
 %immutable ePythonMessagePump::recv_msg;
 %immutable eDVBLocalTimeHandler::m_timeUpdated;
+%immutable eTuxtxtApp::appClosed;
+%immutable eTuxtxtApp::dataAvail;
+%immutable eTuxtxtApp::dataSent;
 %include <lib/base/message.h>
 %include <lib/base/nconfig.h>
 %include <lib/driver/rc.h>
@@ -199,6 +203,7 @@ typedef long time_t;
 %include <lib/dvb/pmt.h>
 %include <lib/components/scan.h>
 %include <lib/components/file_eraser.h>
+%include <lib/components/tuxtxtapp.h>
 %include <lib/driver/avswitch.h>
 %include <lib/driver/rfmod.h>
 %include <lib/driver/misc_options.h>
@@ -303,6 +308,15 @@ void setTunerTypePriorityOrder(int order)
 }
 %}
 
+void setEnableTtCachingOnOff(int);
+%{
+void setEnableTtCachingOnOff(int onoff)
+{
+	eTuxtxtApp *tt = eTuxtxtApp::getInstance();
+	if (tt) tt->setEnableTtCachingOnOff(onoff);
+}
+%}
+
 /************** temp *****************/
 
 	/* need a better place for this, i agree. */
diff --git a/main/Makefile.am b/main/Makefile.am
index 891b646..fed03bf 100644
--- a/main/Makefile.am
+++ b/main/Makefile.am
@@ -53,7 +53,7 @@ enigma2_LDADD = \
 	@LIBUNGIF_LIBS@ \
 	@XML2_LIBS@ \
 	@XMLCCWRAP_LIBS@ \
-	-ldl -lpthread -lcrypt -lresolv -lrt
+	-ldl -lpthread -lcrypt -lresolv -lrt -ltuxtxt32bpp
 
 enigma2$(EXEEXT): $(enigma2_OBJECTS) $(enigma2_DEPENDENCIES) $(enigma2_LDADD_WHOLE)
 #	@rm -f enigma2$(EXEEXT)
